SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialReplicationGraph.cpp

```diff
@@ -86,4 +135,185 @@ void USpatialReplicationGraph::PostReplicateActors(UNetReplicationGraphConnectio
+	// - add dependent Actors
+	TArray<AActor*> ClientInterestedActors =
+		ExtractClientInterestActorsFromGather(ConnectionManager, GatheredReplicationListsForConnection, Viewer);
+
+	return ClientInterestedActors;
+}
+
+TArray<AActor*> USpatialReplicationGraph::ExtractClientInterestActorsFromGather(
+	UNetReplicationGraphConnection* ConnectionManager, FGatheredReplicationActorLists& GatheredReplicationListsForConnection,
+	FNetViewerArray& Viewers)
+{
+	QUICK_SCOPE_CYCLE_COUNTER(NET_ClientEntityInterest_ExtractFromGatheredLists);
+
+	TArray<AActor*> ClientInterestedActors{};
+
+	FPerConnectionActorInfoMap& ConnectionActorInfoMap = ConnectionManager->ActorInfoMap;
+	const uint32 FrameNum = GetReplicationGraphFrame();
+
+	TArray<AActor*> SortingArray{};
+	// Should try and reserve accurately here - could sum gathered rep list lengths (although this wouldn't factor dependent Actors) or
+	// cache the last list size
+
+	for (const FActorRepListConstView& List : GatheredReplicationListsForConnection.GetLists(EActorRepListTypeFlags::Default))
+	{
+		for (AActor* Actor : List)
+		{
+			if (!Actor->GetClass()->HasAnySpatialClassFlags(SPATIALCLASS_SpatialType))
+			{
+				UE_LOG(LogSpatialReplicationGraph, Error,
+					   TEXT("Attempting to add client interest in non-spatial actor (%s), fix class routing"), *Actor->GetName());
+				continue;
+			}
+
+			if (Actor->GetClass()->HasAnySpatialClassFlags(SPATIALCLASS_ServerOnly))
+			{
+				continue;
+			}
+
+			FConnectionReplicationActorInfo& ConnectionActorInfo = ConnectionActorInfoMap.FindOrAdd(Actor);
+
+			// Following comment and logic is from the replication flow - seems maybe relevant to do the same for interest?
+			// Always skip if we've already processed this Actor (because it was in multiple replication lists)
+			if (ConnectionActorInfo.LastInterestUpdateFrameName == FrameNum)
+			{
+				continue;
+			}
+			ConnectionActorInfo.LastInterestUpdateFrameName = FrameNum;
+
+			FGlobalActorReplicationInfo& GlobalActorInfo = GlobalActorReplicationInfoMap.Get(Actor);
+
+			// At this point the replication flow checks ReadyForNextReplication
+			// We avoid because we want interest to remain consistent vs. a per frame replication round robin.
+
+			// Do we run explicit NCD calculations run per Actor?
+			if (bUseNarrowPhaseNCDInterestCulling)
+			{
+				QUICK_SCOPE_CYCLE_COUNTER(NET_ClientEntityInterest_NarrowPhaseDistanceCalculation);
+
+				// Multiple viewers exists in native Unreal to accommodate local multiplayer.
+				int32 ViewersThatSkipActor = 0;
+				for (const FNetViewer& CurViewer : Viewers)
+				{
+					const float DistSq = (GlobalActorInfo.WorldLocation - CurViewer.ViewLocation).SizeSquared();
+					if (ConnectionActorInfo.GetCullDistanceSquared() > 0.f && DistSq > ConnectionActorInfo.GetCullDistanceSquared())
+					{
+						++ViewersThatSkipActor;
+						break;
+					}
+				}
+
+				if (ViewersThatSkipActor >= Viewers.Num())
+				{
+					continue;
+				}
```

Comment:
```suggestion
				int32 ViewersThatSkipActor = 0;
				const float ActorCullDistSq = ConnectionActorInfo.GetCullDistanceSquared();
				if (ActorCullDistSq  <= 0.f)
				{
				    ViewersThatSkipActor = Viewers.Num();
				}
				else
				{
				    for (const FNetViewer& CurViewer : Viewers)
				    {
					    const float DistSq = (GlobalActorInfo.WorldLocation - CurViewer.ViewLocation).SizeSquared();
					    if (DistSq > ActorCullDistSq )
					    {
						    ++ViewersThatSkipActor;
						    break;
					    }
				}

				if (ViewersThatSkipActor >= Viewers.Num())
				{
					continue;
				}
```

Reply:
For bonus point this could be turned into immidiatley executing lambda and ViewersThatSkipActor can be made const that way.

Reply:
Viewers should only ever be one here, so I've refactored the code to account for that.